modules/oe_theme_content_event/src/Plugin/ExtraField/Display/RegistrationDateAwareExtraFieldBase.php

```diff
@@ -0,0 +1,118 @@
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Base class for fields that conditionally render on the registration period.
+ */
+abstract class RegistrationDateAwareExtraFieldBase extends EventExtraFieldBase {
+
+  /**
+   * Current request time, as a timestamp.
+   *
+   * @var int
+   */
+  protected $requestTime;
+
+  /**
+   * Current request time, as a DateTime object.
+   *
+   * @var \DateTimeInterface
+   */
+  protected $requestDateTime;
+
+  /**
+   * RegistrationDateAwareExtraFieldBase constructor.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Component\Datetime\TimeInterface $time
+   *   The time service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_type_manager, TimeInterface $time) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $entity_type_manager);
+    $this->requestTime = $time->getRequestTime();
+    $this->requestDateTime = (new \DateTime())->setTimestamp($this->requestTime);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager'),
+      $container->get('datetime.time')
+    );
+  }
+
+  /**
+   * Apply max-age depending from the registration period time interval.
+   *
+   * @param array $build
+   *   Render array to apply the max-age to.
+   * @param \Drupal\oe_content_event\EventNodeWrapperInterface $event
+   *   Event wrapper object.
+   */
+  protected function applyRegistrationDatesMaxAge(array &$build, EventNodeWrapperInterface $event): void {
+    $cacheable = CacheableMetadata::createFromRenderArray($build);
+    $cacheable->addCacheContexts(['timezone']);
+
+    // Do nothing if the registration is closed.
+    if ($event->isRegistrationClosed($this->requestDateTime)) {
+      $cacheable->applyTo($build);
+      return;
+    }
+
+    // Set start date time interval as max-age if registration is yet to come.
+    if ($event->isRegistrationPeriodYetToCome($this->requestDateTime)) {
+      $cacheable->setCacheMaxAge($event->getRegistrationStartDate()->getTimestamp() - $this->requestTime);
+    }
+
+    // Set end date time interval as max-age if registration is in progress.
+    if ($event->isRegistrationPeriodActive($this->requestDateTime)) {
+      $cacheable->setCacheMaxAge($event->getRegistrationEndDate()->getTimestamp() - $this->requestTime);
```

Comment:
We have to be sure that we have value -1 or > 0.

Reply:
This method will be removed in the next issue OPENEUROPA-2774, I'll remove it from here.

Reply:
I'll actually leave it here, it's been already removed in the next PR.